code/game/objects/items.dm

```diff
@@ -815,3 +820,9 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /// Called on cyborg items that need special charging behavior. Override as needed for specific items.
 /obj/item/proc/cyborg_recharge(coeff = 1, emagged = FALSE)
 	return
+
+/obj/item/proc/retract_to_linked_implant()
+	SIGNAL_HANDLER_DOES_SLEEP
+	if(istype(linked_implant, /obj/item/organ/internal/cyberimp/arm))
+		var/obj/item/organ/internal/cyberimp/arm/arm_implant = linked_implant
+		arm_implant.retract_and_show_radial()
```

Comment:
```
/// Signifies that this proc is used to handle signals, but also sleeps.
/// Do not use this for new work.
#define SIGNAL_HANDLER_DOES_SLEEP

```
try wrapping it in an INVOKE_ASYNC instead.

Reply:
invoke_async added